tools/release/adot-operator-images-mirror/ecr_manager.go

```diff
@@ -0,0 +1,84 @@
+package main
+
+import (
+	"context"
+	"log"
+
+	"github.com/aws/aws-sdk-go-v2/service/ecrpublic"
+	"github.com/cenkalti/backoff"
+)
+
+type ecrManager struct {
+	client       *ecrpublic.Client
+	repositories map[string]bool
+}
+
+func (e *ecrManager) exists(name string) bool {
+	if _, ok := e.repositories[name]; ok {
+		return true
+	}
+
+	return false
+}
+
+func (e *ecrManager) ensure(name string) error {
+	if e.exists(name) {
+		return nil
+	}
+
+	return e.create(name)
```

Comment:
Since we're creating Production ECR Repo. Do you think we can add some name edit check to make sure we can arbitrary create some test repo name in Prod ECR namespace?

Reply:
Sorry I'm not sure if I understand this correctly. By name edit check, do you mean that we need to check if the registry name is as expected (e.g. `adot-operator`)? Thank you!